modulemd/modulemd-defaults.c

```diff
@@ -524,6 +524,60 @@ modulemd_defaults_new_from_string (const gchar *yaml_string, GError **error)
 }
 
 
+/**
+ * modulemd_defaults_new_from_stream:
+ * @yaml_file: A YAML stream containing the module metadata and other related
+ * information such as default streams.
+ * @error: (out): A #GError containing additional information if this function
+ * fails.
+ *
+ * Constructs a new #ModulemdDefaults object from the first valid
+ * modulemd-defaults document in the given module stream. This will ignore any
+ * documents of other types, malformed documents and defaults that appear later
+ * in the stream.
+ *
+ * Returns: A #ModulemdDefaults object constructed the first valid
+ * modulemd-defaults document in the given module stream. This must be freed
+ * with g_object_unref() when no longer needed.
+ *
+ * Since: 1.4
+ */
+ModulemdDefaults *
+modulemd_defaults_new_from_stream (FILE *stream, GError **error)
+{
+  GObject *object = NULL;
+  GPtrArray *data = NULL;
+  ModulemdDefaults *defaults = NULL;
+
+  if (!parse_yaml_stream (stream, &data, error))
+    {
+      return NULL;
+    }
+
+  for (gsize i = 0; i < data->len; i++)
```

Comment:
shouldn't it be `guint`? https://developer.gnome.org/glib/stable/glib-Pointer-Arrays.html#GPtrArray

Reply:
It's irrelevant, really. gsize is essentially size_t which is pretty much designed for this purpose. It's also signed and always going to be at least as large as guint.